src/base/iconprovider.cpp

```diff
@@ -29,6 +29,21 @@
 
 #include "iconprovider.h"
 
+#include <array>
+
+#include <QFileInfo>
+
+namespace
+{
+    std::array<QLatin1String, 5> themeDirs = {
```

Comment:
Why did you split icons into subdirectories?

Reply:
I had to create at least one subdir to follow [the specification](https://standards.freedesktop.org/icon-theme-spec/icon-theme-spec-latest.html). So I decided to follow it completely and use the standard subdirs.

Reply:
`const std::array<QLatin1String, 5> THEME_DIRS`

Reply:
Don't you mind that they will be converted into **QString** many times?